docs/architecture/adr-010-posts-module.md

```diff
@@ -0,0 +1,345 @@
+Each post MUST always have an _author_ which identifies the user that has created the content. It also MUST always reference a _subspace_ inside which it is created. In order to be valid, a post MUST either contain a _text_ or at least one _attachment_.
+
+Optionally, a post can also have a series of _entities_ that have been parsed out of the text. These allow to identify particular content that should be displayed in custom ways (hashtags, mentions, urls, etc).
+
+#### Post
+
+```protobuf
+// Post contains all the information about a single post
+message Post {
+  // Id of the subspace inside which the post has been created 
+  required uint64 subspace_id = 1;
+
+  // Unique id of the post
+  required uint64 id = 2;
+
+  // Text of the post
+  optional string text = 3;
+
+  // Author of the post
+  required string author = 4;
+
+  // Id of the original post of the conversation
+  optional uint64 conversation_id = 5 [default = 0];
+
+  // Reply settings of this post
+  required ReplySetting reply_settings = 6;
+
+  // Creation date of the post
+  required google.protobuf.Timestamp creation_date = 7;
+
+  // Entities connected to this post
+  optional Entities entities = 8;
+}
+
+// Contains the details of entities parsed out of the post text
+message Entities {
+  repeated Tag hashtags = 1;
+  repeated Tag mentions = 2;
+  repeated Url urls = 3;
+}
+
+// ReplySetting contains the possible reply settings that a post can have
+enum ReplySetting {
+  EVERYONE = 1;
+  FOLLOWERS = 2;
+  MENTIONS = 3;
+}
+
+// Tag represents a generic tag 
+message Tag {
+  required uint64 start = 1;
+  required uint64 end = 2;
+  required string tag = 3;
+}
+
+// Url contains the details of a generic URL
+message Url {
+  required uint64 start = 1;
+  required uint64 end = 2;
+  required string url = 3;
+  optional string display_url = 4;
+}
+```
+
+#### Attachments
+
+```protobuf
+// Attachment contains the data of a single post attachment
+message Attachment {
+  required uint32 id = 1;
+  oneof sum {
+    Poll poll = 2;
+    Media media = 3;
+  }
+}
+
+// Media represents a media attachment
+message Media {
+  required string uri = 2;
+  required string mime_type = 3;
+}
+
+// Poll represents a poll attachment
+message Poll {
+  // Question of the poll
+  required string question = 2;
+  
+  // Answers the users can choose from
+  repeated ProvidedAnswer provided_answers = 3;
+  
+  // Date at which the poll will close
+  required google.protobuf.Timestamp end_date = 4;
+  
+  // Whether the poll allows multiple choices from the same user or not 
+  optional bool allows_multiple_answers = 5 [default = false];
+  
+  // Whether the poll allows to edit an answer or not
+  optional bool allows_answer_edits = 6 [default = false];
+  
+  // Provided answer contains the details of a possible poll answer 
+  message ProvidedAnswer {
+    // Text of the answer
+    optional string text = 1;
+    
+    // Attachments of the answer
+    repeated Attachment attachments = 2;
```

Comment:
It allows users to answer a poll with another poll. Should we limit it as `Media` instead?

Reply:
I think that allowing a user to answer a poll with another poll does not have any downside. This might indeed be a wanted feature (i.e. inside a forum)